core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -61,6 +61,15 @@ public ReplicationRequest(ActionRequest request) {
     }
 
     /**
+     * Creates a new request with resolved shard id
+     */
+    public ReplicationRequest(ActionRequest request, ShardId shardId) {
+        super(request);
+        this.index = shardId.getIndex();
+        this.internalShardId = shardId;
```

Comment:
I renamed internalShardId to resolvedShardId and added javadocs to clarify how it is used in https://github.com/areek/elasticsearch/commit/8e7d9b5894e569a47fa29a23d8d193e1aac3ffe9